src/core/gui/dialog/SettingsDialogPaletteTab.cpp

```diff
@@ -0,0 +1,193 @@
+        renderNoPaletteFoundDisclaimer(lb);
+        return;
+    }
+
+    for (const fs::path& p: allPaletteFilePaths) {
+        GtkWidget* listBoxRow = renderPaletteListBoxRow(lb, p);
+
+        if (p == currentlySetPalettePath)
+            gtk_list_box_select_row(GTK_LIST_BOX(lb), GTK_LIST_BOX_ROW(listBoxRow));
+    }
+}
+
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Private Methods
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+auto SettingsDialogPaletteTab::renderPaletteListBoxRow(GtkListBox* lb, const fs::path& p) -> GtkWidget* {
+    Palette palette{p};
+    GtkWidget* listBoxRow = nullptr;
+
+    try {
+        palette.load();
+        listBoxRow = newPaletteListBoxRow(palette);
+    } catch (const std::exception& e) {
+        listBoxRow = newErrorListBoxRow(p, e.what());
+    }
+    setGObjectPalettePath(G_OBJECT(listBoxRow), p);
+
+    gtk_list_box_prepend(lb, listBoxRow);
+    return listBoxRow;
+}
+
+void SettingsDialogPaletteTab::renderColorPaletteExplainLabel() const {
+    gtk_label_set_label(colorPaletteExplainLabel,
+                        FS(_F("<i>The palettes shown below are obtained from:\n"
+                              "   - Built-in palettes: {1}\n"
+                              "   - User palettes: {2}.\n</i>") %
+                           pathLink(Util::getPalettePath()) % pathLink(Util::getConfigSubfolder("palettes")))
+                                .c_str());
+    gtk_label_set_use_markup(colorPaletteExplainLabel, true);
+}
+
+void SettingsDialogPaletteTab::renderNoPaletteFoundDisclaimer(GtkListBox* lb) {
+    GtkWidget* label = gtk_label_new("<span foreground=\"red\">"
+                                     "No palette files (i.e. with extension .gpl) could be found.\n"
+                                     "Using the default until another palette is configured."
+                                     "</span>");
+    gtk_label_set_use_markup(GTK_LABEL(label), true);
+    gtk_widget_show(label);
+    gtk_list_box_prepend(lb, label);
+}
+
+// use list of fs::path as input
+void SettingsDialogPaletteTab::setAllPaletteFilePaths(const std::vector<fs::path>& paletteDirectories) {
+    for (const fs::path& paletteDirectory: paletteDirectories) {
+        std::vector<fs::path> const files = Util::listFilesSorted(paletteDirectory);
+        for (const fs::path& paletteFile: files) {
+            if (paletteFile.extension() == ".gpl") {
+                allPaletteFilePaths.push_back(paletteFile);
+            }
+        }
+    }
+}
+
+auto SettingsDialogPaletteTab::getSelectedPalette() -> std::optional<fs::path> {
+    GtkListBoxRow* selected_listbox_row = gtk_list_box_get_selected_row(paletteListBox);
+    if (allPaletteFilePaths.empty()) {
+        return std::nullopt;
+    }
+    if (!G_IS_OBJECT(selected_listbox_row)) {
+        return std::nullopt;
```

Comment:
```suggestion
        return {};
    }
    if (!G_IS_OBJECT(selected_listbox_row)) {
        return {};
```
Unfortunately, std::optional does not support an overload, to just use the state of the type in it.